src/routes/ask/+page.svelte

```diff
@@ -116,18 +116,25 @@
         });
 
         return async ({ update, result }) => {
-          // if(result.status !== 200) { console.error"("Failed") }
-
           isCreating = false;
 
           await update();
 
-          // @ts-expect-error
-          let message = result.status === 200 ? result.data?.message.content : null;
-
-          if (message) {
-            addMessageToArray(message);
-            determineMagicButton(message);
+          // TODO - Add type checking.
+          //@ts-ignore
+          console.log(result.data.message);
+
+          addMessageToArray({
+            // @ts-ignore
+            role: result.data.message.role,
+            // @ts-ignore
+            content: result.data.message.content
+          });
+
+          // @ts-ignore
+          if (result.data.success) {
+            // @ts-ignore
+            determineMagicButton(result.data.message);
```

Comment:
N2S: stuff like this is what I'm talking about in #48 
It's difficult to know when it is type ActionResult or ActionFailure, and there's really no documentation that I can find on this.